### PR TITLE
feat(sunburst-chart): Add ariaLabel and responsiveness

### PR DESCRIPTION
**Pull request by subarroca**
Feature

- Added aria labels
- Changed overlay to avoid flickering
- Add responsiveness to get more consistent label sizes
- Move central label to front and enable multiline

#### Checklist

- [x] I have read the CONTRIBUTING doc and I follow the PR guidelines
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by subarroca** [comment]
closes #1173

File: apps/components-e2e/src/components/sunburst-chart/sunburst-chart.po.ts [hunk 1]
```diff
@@ -18,18 +18,18 @@ import { Selector } from 'testcafe';
...
 export const centralValue = Selector('.dt-sunburst-chart-selected-value');
 
 export const absoluteBtn = Selector('#chart-value-mode-absolute');
 export const percentBtn = Selector('#chart-value-mode-percent');
 
-export const segments = Selector('.dt-sunburst-chart-group');
+export const segments = Selector('[dt-sunburst-chart-segment]');
 export const sliceShephard = Selector(
-  '.dt-sunburst-chart-group:nth-of-type(5) .dt-sunburst-chart-slice',
+  '[dt-sunburst-chart-segment]:nth-of-type(5) .dt-sunburst-chart-slice',
```
Comment (tomheller) [inline apps/components-e2e/src/components/sunburst-chart/sunburst-chart.po.ts:31]: Could you please change the `nth-of-type` selector to the `nth` helper function, like we did for the stacked-series-chart.

File: libs/barista-components/sunburst-chart/src/sunburst-chart.ts [hunk 2]
```diff
@@ -135,27 +139,64 @@ export class DtSunburstChart {
...
 
+  /** AfterContentInit hook */
+  ngAfterContentInit(): void {
+    this._updateDimensions();
+    this._resizer
+      .change()
+      .pipe(
+        // delay to postpone the reflow to the next change detection cycle
+        takeUntil(this._destroy$),
+        delay(0),
```
Comment (tomheller) [inline libs/barista-components/sunburst-chart/src/sunburst-chart.ts:170]: Nitpick: We could also think about using an AnimationFrameScheduler here to not have to hack the `delay(0)`.

File: libs/barista-components/sunburst-chart/src/sunburst-chart.ts [hunk 3]
```diff
@@ -135,27 +139,64 @@ export class DtSunburstChart {
...
+      )
+      .subscribe(() => {
+        this._updateDimensions();
+        this._cdr.markForCheck();
```
Comment (tomheller) [inline libs/barista-components/sunburst-chart/src/sunburst-chart.ts:174]: Our coding guidelines state to not abbreviate variable names if at all possible.
```suggestion
        this._changeDetectorRef.markForCheck();
```

File: libs/barista-components/sunburst-chart/src/sunburst-chart.ts [hunk 4]
```diff
@@ -194,110 +235,47 @@ export class DtSunburstChart {
...
+   * Updates the position of the overlay to create a mouseFollow position
+   */
+  _handleOnMouseMove(event: MouseEvent): void {
     if (this._overlayRef) {
-      this._overlayRef.detach();
-      this._portal = null;
+      this._overlayService._positionStrategy.setOrigin({
+        x: event.clientX,
+        y: event.clientY,
+      });
```
Comment (tomheller) [inline libs/barista-components/sunburst-chart/src/sunburst-chart.ts:266]: Please check this in firefox if you haven't already. We had the problems in the past with clientX or event offsets in general not being the same on svg elements in firefox as in blink-based browsers.

File: libs/barista-components/sunburst-chart/src/sunburst-chart.ts [hunk 5]
```diff
@@ -194,110 +235,47 @@ export class DtSunburstChart {
...
-
-    this._dismissOverlay();
-    if (origin) {
-      //  TODO: lukas.holzer
-      // Bazel cannot resolve the extends of the interface properly and
-      // throws error: Property 'data' does not exist on type 'DtSunburstChartSlice'.
-      // @ts-ignore
-      this._createOverlay(origin, node.data);
-    }
-  }
```
Comment (tomheller) [inline libs/barista-components/sunburst-chart/src/sunburst-chart.ts:214]: If you deprecate the code, you cannot actually delete it. It still needs to work. 
Why are you removing this functionality exactly? Nothing was stated in the issue that you are planning to remove this.

File: libs/barista-components/sunburst-chart/src/sunburst-chart.ts [hunk 6]
```diff
@@ -194,110 +235,47 @@ export class DtSunburstChart {
       this._selectedRelativeValue = 1;
     }
   }
-  /** Programmatically close the overlay. */
-  closeOverlay(): void {
-    this._dismissOverlay();
-  }
```
Comment (tomheller) [inline libs/barista-components/sunburst-chart/src/sunburst-chart.ts:200]: If you deprecate the code, you cannot actually delete it. It still needs to work.
Why are you removing this functionality exactly? Nothing was stated in the issue that you are planning to remove this.

File: libs/barista-components/sunburst-chart/src/sunburst-chart.ts [hunk 7]
```diff
@@ -135,27 +139,64 @@ export class DtSunburstChart {
...
     private _platform: Platform,
     // TODO: remove this sanitizer when ivy is no longer opt out
     private _sanitizer: DomSanitizer,
-    private _elementRef?: ElementRef<HTMLElement>,
-    @Optional()
-    @Inject(DT_UI_TEST_CONFIG)
-    private _config?: DtUiTestConfiguration,
+    private _elementRef: ElementRef<HTMLElement>,
+    private _resizer: DtViewportResizer,
+    private _cdr: ChangeDetectorRef,
```
Comment (tomheller) [inline libs/barista-components/sunburst-chart/src/sunburst-chart.ts:159]: Could we do this without a breaking change as well? When making the new parameters optional and have a fallback to the way that it is currently working, we should be able to do this without a breaking change. 
As I assume that after we released version 7 a couple of weeks ago, version 8 will take some time to be released.

File: libs/barista-components/sunburst-chart/src/sunburst-chart.ts [hunk 8]
```diff
@@ -49,13 +49,10 @@ import {
...
-import { Platform } from '@angular/cdk/platform';
-import { DomSanitizer, SafeStyle } from '@angular/platform-browser';
-import { DtColors } from '@dynatrace/barista-components/theming';
 
-const OVERLAY_PANEL_CLASS = 'dt-sunburst-chart-overlay-panel';
-const VIEWBOX_WIDTH = 480;
-const VIEWBOX_HEIGHT = 400;
+/** Minimum width of the chart */
+const MIN_WIDTH = 480;
+const MAX_WIDTH = 800;
```
Comment (ffriedl89) [inline libs/barista-components/sunburst-chart/src/sunburst-chart.ts:55]: Why do we need a max width here? The surrounding layout should limit the chart from growing.

File: libs/barista-components/sunburst-chart/src/sunburst-chart.ts (hunk 7 above)
Comment (ffriedl89) [reply libs/barista-components/sunburst-chart/src/sunburst-chart.ts:159]: Yeah please leave even unused parameters in there to avoid breaking changes. Version 8 is far out. No plans or timeline yet.

File: libs/barista-components/sunburst-chart/README.md [hunk 9]
```diff
@@ -47,10 +48,10 @@ attribute is required to create a valid output.
 
 #### Methods
 
-| Name           | Params                              | Description                         |
-| -------------- | ----------------------------------- | ----------------------------------- |
-| `openOverlay`  | `node: DtSunburstChartTooltipNode>` | Open an overlay for the given node. |
-| `closeOverlay` |                                     | Closes the overlay if open.         |
+| Name           | Params                              | Description                                     |
+| -------------- | ----------------------------------- | ----------------------------------------------- |
+| `openOverlay`  | `node: DtSunburstChartTooltipNode>` | Deprecated. Open an overlay for the given node. |
```
Comment (ffriedl89) [inline libs/barista-components/sunburst-chart/README.md:53]: Why deprecate this method? It seems rather useful to have a possibility to open the overlay programmatically.

File: libs/barista-components/sunburst-chart/src/sunburst-chart.ts (hunk 4 above)
Comment (subarroca) [reply libs/barista-components/sunburst-chart/src/sunburst-chart.ts:266]: works as a charm :D

File: libs/barista-components/sunburst-chart/src/sunburst-chart.ts (hunk 7 above)
Comment (subarroca) [reply libs/barista-components/sunburst-chart/src/sunburst-chart.ts:159]: @tomheller @ffriedl89 I can keep old params but for new params we're gonna loose the ability of scaling the chart on resize. I think it would be good to open a new PR for this so I'll leave the code commented

File: libs/barista-components/sunburst-chart/src/sunburst-chart-segment.ts [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Component, Input } from '@angular/core';
+import { DtSunburstChartNodeSlice } from './sunburst-chart.util';
+
+@Component({
+  selector: 'svg:g[dt-sunburst-chart-segment]',
+  templateUrl: 'sunburst-chart-segment.html',
```
[the remark this answers is not in the document]
Comment (subarroca) [reply libs/barista-components/sunburst-chart/src/sunburst-chart-segment.ts:22]: Tests fail when using svg file instead of html

**Comment by tomheller** [comment]
@subarroca 
Seems that there is still a build error, one of the functions called from the template has a parameter mismatch

```
------------------------------------------------------------------------------
 
Building entry point '@dynatrace/barista-components/sunburst-chart'
 
------------------------------------------------------------------------------
 
Compiling TypeScript sources through ngc
 
ERROR: libs/barista-components/sunburst-chart/src/sunburst-chart.html(14,5): Expected 0 arguments, but got 1.
 ```

File: libs/barista-components/sunburst-chart/src/sunburst-chart.ts [hunk 11]
```diff
@@ -135,27 +112,133 @@ export class DtSunburstChart {
...
-  private _overlayRef: OverlayRef;
+  private _overlayRef: DtOverlayRef<DtSunburstChartTooltipData> | null;
+
+  /** @internal Destroy subject to clear subscriptions on component destroy. */
+  private readonly _destroy$ = new Subject<void>();
 
   constructor(
-    private _overlayService: Overlay,
-    private _viewContainerRef: ViewContainerRef,
+    private _overlayService: DtOverlay,
```
Comment (tomheller) [inline libs/barista-components/sunburst-chart/src/sunburst-chart.ts:168]: I guess this would still be a breaking change, when changing the type from `Overlay` to `DtOverlay`.

File: libs/barista-components/sunburst-chart/src/sunburst-chart.ts [hunk 12]
```diff
@@ -135,27 +112,133 @@ export class DtSunburstChart {
...
+    _viewContainerRef: ViewContainerRef,
     private _platform: Platform,
     // TODO: remove this sanitizer when ivy is no longer opt out
     private _sanitizer: DomSanitizer,
-    private _elementRef?: ElementRef<HTMLElement>,
+    private _elementRef: ElementRef<HTMLElement>,
+    // private _resizer: DtViewportResizer,
+    // private _changeDetectorRef: ChangeDetectorRef,
```
Comment (tomheller) [inline libs/barista-components/sunburst-chart/src/sunburst-chart.ts:175]: We can make the parameters provided by the DI optional. This way, if the component is used in the template, everything will work as expected. The breaking change in the constructor usually only affects consumers, that would create components programatically using a componentFactory. 

```suggestion
    /** @breaking-change Make the DtViewportResizer mandatory in version 8.0.0 */
    @Optional() private _resizer?: DtViewportResizer,
    /** @breaking-change Make the ChangeDetectorRef mandatory in version 8.0.0 */
    @Optional() private _changeDetectorRef?: ChangeDetectorRef,
```

File: libs/barista-components/sunburst-chart/src/sunburst-chart.ts [hunk 13]
```diff
@@ -135,27 +112,133 @@ export class DtSunburstChart {
...
     @Optional()
     @Inject(DT_UI_TEST_CONFIG)
-    private _config?: DtUiTestConfiguration,
+    _config?: DtUiTestConfiguration,
   ) {}
 
+  /** AfterContentInit hook */
+  // TODO: Uncomment this code to allow resizing.
+  // It's commented now to avoid a breaking change
```
Comment (tomheller) [inline libs/barista-components/sunburst-chart/src/sunburst-chart.ts:183]:
```suggestion
```

File: libs/barista-components/sunburst-chart/src/sunburst-chart.ts [hunk 14]
```diff
@@ -135,27 +112,133 @@ export class DtSunburstChart {
...
+    // this._resizer
+    //   .change()
+    //   .pipe(
+    //     takeUntil(this._destroy$),
+    //   )
+    //   .subscribe(() =>
+    //     animationFrameScheduler.schedule(() => {
+    //       this._updateDimensions();
+    //       this._changeDetectorRef.markForCheck();
+    //     }, 0, 0));
```
Comment (tomheller) [inline libs/barista-components/sunburst-chart/src/sunburst-chart.ts:196]:
```suggestion
// @breaking-change Remove the if check when the resizer is no longer optional
if(this._resizer) {
     this._resizer
       .change()
       .pipe(
         takeUntil(this._destroy$),
       )
       .subscribe(() =>
         animationFrameScheduler.schedule(() => {
           this._updateDimensions();
           this._changeDetectorRef.markForCheck();
         }, 0, 0));
}
```

**Comment by tomheller** [comment]
@subarroca We don't need to uncomment the breaking change, but we can make the changes in the constructor optional, and within the component deal with the fact that values provided by the DI might not be there (i.e. the component should not break if `this._resizer` is undefined). 

This way we can roll out the functionality to most consumers without breaking the components api and progressively enhance the features.

**Comment by ffriedl89** [comment]
cherry pick done
